js/api.js

```diff
@@ -0,0 +1,64 @@
+async function fetchTodosFromAPI() {
+    try {
+        const response = await fetch('https://dummyjson.com/todos');
+        if (!response.ok) {
+            throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+
+        const { todos } = await response.json();
+
+        if (!Array.isArray(todos)) {
+            console.error('Invalid response structure');
+            return [];
+        }
+
+        return todos;
+    } catch (error) {
+        console.error('Error fetching todos:', error);
+        return [];
+    }
+}
+
+async function apiRequest(url, method, body) {
+    try {
+        const options = {
+            method,
+            ...(method !== 'DELETE' ? {
+                headers: { 'Content-Type': 'application/json' },
+                body: JSON.stringify(body)
+            } : {})
+        };
+
+        const response = await fetch(url, options);
+        return response.ok;
+    } catch (error) {
+        console.error(`Error during ${method} request:`, error);
+        return false;
+    }
+}
+
+function addTodoToAPI(todo) {
+    const requestBody = {
+        todo: todo.todo,
+        completed: todo.completed,
+        userId: todo.userId,
+    };
+    return apiRequest('https://dummyjson.com/todos/add', 'POST', requestBody);
+}
+
+function deleteTodoFromAPI(todoId) {
+    return apiRequest(`https://dummyjson.com/todos/${todoId}`, 'DELETE');
+}
+
+function updateTodoInAPI(updatedTodo, updateField) {
+    const requestBody = {};
+
+    if (updateField === 'completed') {
```

Comment:
Still not recongized what is the need for updated field since you are sending the needed data in the task object for example
```
task.completed = !task.completed;
const isUpdated = await updateTodoInAPI(task, 'completed');
```

And the same here

```
task.todo = updatedText;
const isUpdated = await updateTodoInAPI(task, 'todo');
```

Reply:
There are only two things that can be updated on the task itself (the completed state, to-do content), When I read the API documentation I saw that they send a single field at a time. For example, the completed field with its value alone and so on. So I made it like what they did, also the user updates either the completed status at a time and the request will be sent for updating the completed status, or either for updating the todo but not both at the same time, so I used it like this (to not send both fields in each request while only one is updated, got it?)

It is not necessary by the way, so I refactored it to send the completed and todo fields each time as you suggested.